src/main/resources/assets/react/documentTypes/NewOrEditDocumentType.jsx

```diff
@@ -104,8 +107,42 @@ export function NewOrEditDocumentType({
 			});
 	}
 
+	/**
+	 * Gets all the names of existing documentTypes
+	 *
+	 */
+	function getDocumentTypeNames() {
+		fetch(`${servicesBaseUrl}/graphQL`, {
```

Comment:
The list of documentTypes are already fetched by a parent component, can't you simply pass it in?